src_c/event.c

```diff
@@ -1964,6 +1990,12 @@ pg_event_get(PyObject *self, PyObject *args, PyObject *kwargs)
 
     _pg_event_pump(dopump);
 
+    int i;
+    for (i = 0; i < NUM_KEYS; i++) {
+        memset(pressed_keys, 0, sizeof(pressed_keys));
+        memset(released_keys, 0, sizeof(released_keys));
+    }
```

Comment:
This for loop zeroes the whole array out 512 times in a loop. The loop is unnecessary.

I believe sizeof(pressed_keys) will return the total bytes of the array because it's defined "statically", it has a set size.

You can memset over the whole thing at once.